Core/clim-basic/geometry/region-composition.lisp

```diff
@@ -77,7 +63,7 @@
     :regions (append (standard-region-set-regions a)
                      (standard-region-set-regions b))))
 
-(defmethod region-union ((a region) (b region))
+(defmethod region-union ((a bounding-rectangle) (b bounding-rectangle))
```

Comment:
The commit message says

>"General" methods specialized on region have been changed to specialize on a bounding-rectangle to avoid ambiguity.

Can you give an example? The method bodies don't seem to require the region to be a `bounding-rectangle`.

Reply:
all unbounded regions are handled in regions.lisp (see macro def-method). If we specialize "just" on region, then the right method made specially for unbound regions may be replaced by a another one. For instance:

```
(DEFMETHOD REGION-DIFFERENCE ((A REGION) (B EVERYWHERE-REGION))
     (DECLARE (IGNORABLE A B))
     +NOWHERE+)

(defmethod region-difference ((a rectangle) (b region))
  (make-instance 'standard-region-difference a b))
```

problem does not occur if we specialize all method on bounded-rectangles. The point is that for all unbounded regions have methods defined and we do not want to replace them.

Reply:
I see. Could you add this explanation to `region-composition.lisp` as a comment and the commit message?

Reply:
I've added it to file regions.lisp (and to commit message), regions lisp is more appropriate place because that's where this requirement comes from and it spans not only region-composition, but also region-predicates and region-set files.